jsstring.c

```diff
@@ -665,7 +665,7 @@ static void Sp_split(js_State *J)
 {
 	if (js_isundefined(J, 1)) {
 		js_newarray(J);
-		js_copy(J, 0);
+		js_pushstring(J, js_tostring(J, 0));
```

Comment:
This change potentially creates a new allocation for the string.

Instead, you could probably just add `js_tostring(J, 0)` before the `js_copy(J, 0)`. This will convert the `this` object into a string, by reusing the existing string (allocated or not) and then `js_copy` does not need to allocate anything.

Reply:
Well, this can be optimized in future, i see no real reason to bother with this right now just for a single method, where other methods are left without any care (especially since lots of them use checkstrings). This is rather a rare case when this conversion is actually necessary. I wouldn't notice the bug if not test262 set. 

IMO the whole MuJS code base is awesome just because it remains so simple and concise. I wouldn't say MuJS really cares that much about memory optimizations in the first place, since it's rather an attempt to try balance both readability and effectivity (strings are unfortunately where mujs fails miserably though). 

So unless you want to bother, go ahead, but i have no interest in that.

Reply:
I agree that mujs is beautiful code and that it lacks in some areas when comes to strings. Your original PR does work even if it does an unnecessary allocation. And yes, I do have fixes for some of the string issues of mujs and I intend to get it in eventually.